### PR TITLE
Fix #6923 Check for queue correctly

### LINKED ISSUE #6923: Queue path tiles get replaced by normal path on mouse-over with path scenery
**OS:** Windows 10
**Version:** 0.1.2
**Commit/Build:** eb4a892

When trying to build foothpath scenery (such as litter bins, benches, lamps or jumping fountains) hovering over a queue path will replace it with a normal footpath.

- [ ] Reproducible in RCT2 (vanilla)? I don't think so, but haven't tried as it doesn't run on my PC :(
- [ ] Multiplayer?

**Steps to reproduce:**
1. Build a few tiles of queue path. It doesn't have to be connected to a ride. Existing paths from a savegame or scenario work as well.
2. Select some foothpath scenery (for example a normal bench) from the scenery window.
3. Move the cursor over the queue path. It will get replaced by a normal foothpath.

pathbug

### PR DESCRIPTION
**Pull request by Broxzier**

### CODE REVIEW

File: src/openrct2/world/footpath.c [hunk 1]
```diff
@@ -270,7 +270,8 @@ static money32 footpath_element_insert(sint32 type, sint32 x, sint32 y, sint32 z
 
 static money32 footpath_element_update(sint32 x, sint32 y, rct_tile_element *tileElement, sint32 type, sint32 flags, uint8 pathItemType)
 {
-    if (footpath_element_get_type(tileElement) != (type & 0x0F) || (tileElement->type & FOOTPATH_ELEMENT_TYPE_FLAG_IS_QUEUE) != (type >> 7)) {
+    if (footpath_element_get_type(tileElement) != (type & (FOOTPATH_PROPERTIES_TYPE_MASK >> 4)) ||
```
Comment (Gymnasiast) [inline src/openrct2/world/footpath.c]: This looks a bit dirty to me. If several values are put together in `type`, perhaps we should split those?

File: src/openrct2/world/footpath.c [hunk 2]
```diff
@@ -270,7 +270,8 @@ static money32 footpath_element_insert(sint32 type, sint32 x, sint32 y, sint32 z
...
+        footpath_element_is_queue(tileElement) != (type >> 7 == 1)) {
```
Comment (duncanspumpkin) [inline src/openrct2/world/footpath.c]: I always get scared with order of execution when using shifts. I would add brackets even if not strictly required.

File: src/openrct2/world/footpath.c [hunk 3]
```diff
@@ -270,7 +270,11 @@ static money32 footpath_element_insert(sint32 type, sint32 x, sint32 y, sint32 z
 
 static money32 footpath_element_update(sint32 x, sint32 y, rct_tile_element *tileElement, sint32 type, sint32 flags, uint8 pathItemType)
 {
-    if (footpath_element_get_type(tileElement) != (type & 0x0F) || (tileElement->type & FOOTPATH_ELEMENT_TYPE_FLAG_IS_QUEUE) != (type >> 7)) {
+    const sint32 newFootpathType = (type & (FOOTPATH_PROPERTIES_TYPE_MASK >> 4));
```
Comment (Gymnasiast) [inline src/openrct2/world/footpath.c]: This will do for now. Eventually, the signature of `footpath_element_{update,insert}` must be modified.

**Comment by duncanspumpkin** [comment]
Shouldn't this bump the network version or is the bug in top toolbar code rather than the footpath code

**Comment by Broxzier** [comment]
Oh yeah, it should. It changes the behaviour.

**Pull request merged**